sql/src/main/java/io/crate/planner/statement/CopyStatementPlanner.java

```diff
@@ -177,10 +184,33 @@ public static ExecutionPlan planCopyFromExecution(DiscoveryNodes allNodes,
             clusteredBy == null ? null : table.getReference(clusteredBy)
         );
         Reference rawOrDoc = rawOrDoc(table, partitionIdent);
-        int rawOrDocIdx = toCollect.size();
+        final int rawOrDocIdx = toCollect.size();
         toCollect.add(rawOrDoc);
+
+        List<? extends Symbol> projectionOutputs = AbstractIndexWriterProjection.OUTPUTS;
+        boolean isReturnSummary = copyFrom instanceof CopyFromReturnSummaryAnalyzedStatement;
+        SourceIndexWriterProjection.ReturnSummarySymbols returnSummarySymbols = null;
+        if (isReturnSummary) {
+            final InputColumn sourceUriSymbol = new InputColumn(toCollect.size(), DataTypes.STRING);
+            toCollect.add(new Reference(
```

Comment:
Maybe the expressions could have a function to get a Reference for the table, to keep the return-type information contained in the expression / close together to the implementation.